elements/tcpudp/fasttcpflows.cc

```diff
@@ -103,9 +108,14 @@ FastTCPFlows::change_ports(int flow)
   tcp->th_dport = dport;
   tcp->th_sum = 0;
   len = _len-14-sizeof(click_ip);
+#if CLICK_LINUXMODULE
   csum = ~click_in_cksum((unsigned char *)tcp, len) & 0xFFFF;
-  tcp->th_sum = csum_tcpudp_magic
-    (_sipaddr.s_addr, _dipaddr.s_addr, len, IP_PROTO_TCP, csum);
+  tcp->th_sum = csum_tcpudp_magic(_sipaddr.s_addr, _dipaddr.s_addr,
+                    len, IP_PROTO_TCP, csum);
+#else
+  csum = click_in_cksum((uint8_t *)tcp, len);
+  tcp->th_sum = click_in_cksum_pseudohdr(csum, ip, len);
+#endif
```

Comment:
So can you do this a bit of a different way? I think in this case and all similar cases you want to just use `click_in_cksum_pseudohdr_raw`. Specifically:

```
csum = click_in_cksum((uint8_t*) tcp, len);
tcp->th_sum = click_in_cksum_pseudohdr_raw(csum, _sipaddr.s_addr, _dipaddr.s_addr, IP_PROTO_TCP, len);
```

On linuxmodule that function is a `#define` for csum_tcpudp_magic.

Note that it would probably be just as fast to use click_in_cksum_pseudohdr, but I don't feel strongly.